pkgs/os-specific/linux/r8152-udev-rules/default.nix

```
@@ -0,0 +1,33 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+}:
+stdenv.mkDerivation rec {
+  pname = "r8152-udev-rules";
+  version = "2.16.3.20221209";
+
+  src = fetchFromGitHub {
+    owner = "wget";
+    repo = "realtek-r8152-linux";
+    rev = "v${version}";
+    sha256 = "sha256-RaYuprQFbWAy8CtSZOau0Qlo3jtZnE1AhHBgzASopSA=";
+  };
+
+  # We don't want to build the kernel module
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/lib/udev/rules.d
+    cp 50-usb-realtek-net.rules $out/lib/udev/rules.d/
+    runHook postInstall
```

```suggestion
    runHook preInstall

    mkdir -p $out/lib/udev/rules.d
    cp 50-usb-realtek-net.rules $out/lib/udev/rules.d/

    runHook postInstall
```
just a formatting nit, and you can add it since you have to touch the file anyways.